dcr.go

```diff
@@ -0,0 +1,55 @@
+package dcrlibwallet
+
+import (
+	"os"
+	"path/filepath"
+
+	"decred.org/dcrwallet/v2/errors"
+
+	"github.com/asdine/storm"
+
+	bolt "go.etcd.io/bbolt"
+
+	"github.com/planetdecred/dcrlibwallet/wallets/dcr"
+)
+
+func initializeDCRWallet(rootDir, dbDriver, netType string) (*storm.DB, string, error) {
```

Comment:
This doesn't really initialize a dcr wallet; the databases initialized here aren't specifically used for a single dcr wallet, one is used to store multiwallet data including ALL (dcr) wallets info and also general, non-wallet specific config data. This is really actually initializing the multiwallet. It also even initializes the log rotator which is not tied to any wallet in particular but is meant for all logging done by this library. I'd say let's move the body of this function back to the `NewMultiwallet` function, and also retain the rootDir without appending "dcr".

Reply:
I suppose we'd need a function like this in the dcr package, perhaps a `InitializeWallet` function so it reads as `dcr.InitializeWallet`. I'm not yet completely sure what the `dcr.InitializeWallet` function would do but it would be quite similar to what we have here. It wouldn't/shouldn't initialize a new log rotator though, the caller of the function should provide a logger to use. It also likely won't need to init a politeia database since politeia data won't be stored per wallet.

Reply:
appending "dcr" enables us keep all dcr database files under a single folder and seems easier to manage/track. I agree the method name is icky, and doesn't properly describe what's going on.

having separate rootdir for different assets looks advantageous in terms of folder structure
![Screenshot from 2022-08-23 19-47-46](https://user-images.githubusercontent.com/25265396/186239395-6d66005c-15f4-4c71-8599-6f3f66923954.png)

also do you think it's remotely possible we might want to be using different database drivers for different coins?

Reply:
> having separate rootdir for different assets looks advantageous in terms of folder structure

I agree. What I was getting at is, the databases created here aren't even for all dcr wallets alone but for all wallets. It's the multiwallet parent database that tracks ALL wallets for all assets, which is used to populate/restore all the asset wallets when multiwallet is initialized. It also stores general config values that aren't related to any wallet.

The "dcr" subfolder should be reserved for dcr wallet databases, as in your screenshot. But there should also exist in the rootDir, a multiwallet.db file (currently called wallets.db, as in storing info for all wallets, which makes sense too).

> do you think it's remotely possible we might want to be using different database drivers for different coins?

Yes. Very possible.